### PR TITLE
feat(pullzone): add additional pullzone fields

### PR DESCRIPTION
**Pull request by jspizziri**

### CODE REVIEW

File: pullzone_get.go [hunk 1]
```diff
@@ -121,13 +129,17 @@ type PullZone struct {
...
 	OriginShieldQueueMaxWaitTime          *int32      `json:"OriginShieldQueueMaxWaitTime,omitempty"`
 	OriginShieldZoneCode                  *string     `json:"OriginShieldZoneCode,omitempty"`
+	OriginType                            *int32      `json:"OriginType,omitempty"`
 	OriginURL                             *string     `json:"OriginUrl,omitempty"`
 	PermaCacheStorageZoneID               *int64      `json:"PermaCacheStorageZoneId,omitempty"`
 	PriceOverride                         *float64    `json:"PriceOverride,omitempty"`
 	QueryStringVaryParameters             []string    `json:"QueryStringVaryParameters,omitempty"`
 	RequestLimit                          *int32      `json:"RequestLimit,omitempty"`
+	ShieldDDosProtectionEnabled           *bool       `json:"ShieldDDosProtectionEnabled,omitempty"`
+	ShieldDDosProtectionType              *int32      `json:"ShieldDDosProtectionType,omitempty"`
```
Comment (fho) [inline pullzone_get.go:139]:
```suggestion
	ShieldDDosProtectionEnabled           *bool       `json:"ShieldDDosProtectionEnabled,omitempty"`
	ShieldDDosProtectionType              *int      `json:"ShieldDDosProtectionType,omitempty"`
```

File: pullzone_get.go [hunk 2]
```diff
@@ -83,9 +85,13 @@ type PullZone struct {
 	IgnoreQueryStrings                    *bool       `json:"IgnoreQueryStrings,omitempty"`
 	LimitRateAfter                        *float64    `json:"LimitRateAfter,omitempty"`
 	LimitRatePerSecond                    *float64    `json:"LimitRatePerSecond,omitempty"`
+	LogAnonymizationType                  *int32      `json:"LogAnonymizationType,omitempty"`
+	LogFormat                             *int32      `json:"LogFormat,omitempty"`
 	LogForwardingEnabled                  *bool       `json:"LogForwardingEnabled,omitempty"`
+	LogForwardingFormat                   *int32      `json:"LogForwardingFormat,omitempty"`
 	LogForwardingHostname                 *string     `json:"LogForwardingHostname,omitempty"`
 	LogForwardingPort                     *int32      `json:"LogForwardingPort,omitempty"`
+	LogForwardingProtocol                 *int32      `json:"LogForwardingProtocol,omitempty"`
```
Comment (fho) [inline pullzone_get.go:94]:
```suggestion
	LogForwardingProtocol                 *int      `json:"LogForwardingProtocol,omitempty"`
```

File: pullzone_get.go [hunk 3]
```diff
@@ -83,9 +85,13 @@ type PullZone struct {
 	IgnoreQueryStrings                    *bool       `json:"IgnoreQueryStrings,omitempty"`
 	LimitRateAfter                        *float64    `json:"LimitRateAfter,omitempty"`
 	LimitRatePerSecond                    *float64    `json:"LimitRatePerSecond,omitempty"`
+	LogAnonymizationType                  *int32      `json:"LogAnonymizationType,omitempty"`
+	LogFormat                             *int32      `json:"LogFormat,omitempty"`
 	LogForwardingEnabled                  *bool       `json:"LogForwardingEnabled,omitempty"`
+	LogForwardingFormat                   *int32      `json:"LogForwardingFormat,omitempty"`
```
Comment (fho) [inline pullzone_get.go:91]:
```suggestion
	LogForwardingFormat                   *int      `json:"LogForwardingFormat,omitempty"`
```

File: pullzone_get.go [hunk 4]
```diff
@@ -83,9 +85,13 @@ type PullZone struct {
 	IgnoreQueryStrings                    *bool       `json:"IgnoreQueryStrings,omitempty"`
 	LimitRateAfter                        *float64    `json:"LimitRateAfter,omitempty"`
 	LimitRatePerSecond                    *float64    `json:"LimitRatePerSecond,omitempty"`
+	LogAnonymizationType                  *int32      `json:"LogAnonymizationType,omitempty"`
```
Comment (fho) [inline pullzone_get.go:88]:
```suggestion
	LogAnonymizationType                  *int      `json:"LogAnonymizationType,omitempty"`
```

**Review by fho — COMMENTED** [review]
1. I can't find the OriginType, OriginHostHeader, EnableSmartCache fields in the API documentation[^1].
Where are they coming from?

2. I suggest that we use the golang `int` type when the API states `int` because this is consistent with how I mapped bunny `int` so far to Golangs types. It also requires less often casting. 
Using better fitting smaller int types (down to uint8) when possible makes also sense but then we would have to make it consistent. 


[^1]: https://docs.bunny.net/reference/pullzonepublic_index2

**Comment by jspizziri** [comment]
@fho 

1. While they're not documented they are present in the physical response of a PullZone. My guess is that their documentation is simply lagging behind their implementation.
2. I've committed your suggestions, squashed, and pushed.

**Comment by fho** [comment]
The Pull-Request is merged.
I cherry-picked the commit manually to main.

**Pull request closed without merging**